src/components/app/styles/App.module.css

```diff
@@ -8,6 +8,12 @@
     isolation: isolate;
 }
 
+/* Avoid vertical layer dialog scroll on small screens */
+:global(#dhis2-portal-root) aside[role=dialog],
+:global(#dhis2-portal-root) aside[role=dialog] > div > div {
+    max-height: calc(100vh - 64px)
+}
+
```

Comment:
There's some ongoing discussion about making the UI `Modal` vertical-padding customisable. This is another use case where it would be useful, so I'm making a note of it 👍